2025-10-31 12:09:31,265 : INFO : root : Loading config.
2025-10-31 12:09:31,270 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2025-10-31 12:09:31,271 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\cfa6ecba65704a931c9dd214c44986c8.mpf_cache
2025-10-31 12:09:31,380 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-10-31 12:09:31,381 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\91cc4d1c993c58dc354e34974b9cfeac.mpf_cache
2025-10-31 12:09:31,382 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f98ffc0ff5962effface7d86bc406d5c.mpf_cache
2025-10-31 12:09:31,382 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\76792f1d686f89709887dba7bd69667a.mpf_cache
2025-10-31 12:09:31,383 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\142bbae4707335cc8e3efaa4364c4893.mpf_cache
2025-10-31 12:09:31,383 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\d586631024bd380382ddba1caafd4910.mpf_cache
2025-10-31 12:09:31,384 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\202647b69b904517343fecdf4579c275.mpf_cache
2025-10-31 12:09:31,387 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\d069db2c83fe896d88972cde9b686138.mpf_cache
2025-10-31 12:09:31,388 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\White\Documents\Top_Gun_Pinball\modes\mig_letters\config\mig_letters.yaml
2025-10-31 12:09:31,388 : INFO : ConfigProcessor : Loading config from file C:\Users\White\Documents\Top_Gun_Pinball\modes\mig_letters\config\mig_letters.yaml.
2025-10-31 12:09:31,407 : INFO : ConfigProcessor : Loading config: C:\Users\White\Documents\Top_Gun_Pinball\modes\mig_letters\config\mig_letters.yaml
2025-10-31 12:09:31,407 : ERROR : ConfigProcessor : Config section 'sounds' is removed in MPF 0.80 and will be ignored.
2025-10-31 12:09:31,408 : ERROR : ConfigProcessor : Config section 'slides' is removed in MPF 0.80 and will be ignored.
2025-10-31 12:09:31,408 : INFO : ConfigProcessor : Config file cache created: C:\Users\White\AppData\Local\Temp\58ab893dda10a3d2ba8fe97b81891df0.mpf_cache
2025-10-31 12:09:31,409 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\0f5f0ac41d1a2672d0d32cef2b12699e.mpf_cache
2025-10-31 12:09:31,420 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\3a419899266be7cd41a083be5a3a8ebb.mpf_cache
2025-10-31 12:09:31,431 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\82e20c7c6490dd2e5fed726a3aa8d652.mpf_cache
2025-10-31 12:09:31,445 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\359114d5ba2a6ad6c93d1559776bbb30.mpf_cache
2025-10-31 12:09:31,459 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f8c9a3739ec88c539117633fa55f5d20.mpf_cache
2025-10-31 12:09:31,471 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\d421369d6df802e4e6f8aa58f446b1a6.mpf_cache
2025-10-31 12:09:31,481 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\2151caa0419aeb1e56d3d8934477383e.mpf_cache
2025-10-31 12:09:31,495 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\17003635873251b4f6841bf4ab3bdcdd.mpf_cache
2025-10-31 12:09:31,507 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\c8ceed3d7c617b2f1fa85b3487b918c6.mpf_cache
2025-10-31 12:09:31,519 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\f70c640a816343cb9040222d7a9269d3.mpf_cache
2025-10-31 12:09:31,536 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\30922048be1d0c17ea15fc75f7fbfe10.mpf_cache
2025-10-31 12:09:31,548 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\2e44e1a8157cd8d5be0e623de5bcf3cd.mpf_cache
2025-10-31 12:09:31,559 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\58d7809d3d969de03aaf131fc45863f8.mpf_cache
2025-10-31 12:09:31,572 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\a2ff6e5c252adf2ba75ad894f77f7ce6.mpf_cache
2025-10-31 12:09:31,582 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\2ebcf623e49efab13a66f36a8282b4e9.mpf_cache
2025-10-31 12:09:31,594 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\d380fa2863f93e197660f0281ffd272b.mpf_cache
2025-10-31 12:09:31,604 : INFO : ConfigProcessor : Loading config from cache: C:\Users\White\AppData\Local\Temp\88e41ae7dcea899a043a2f63da55a8b7.mpf_cache
2025-10-31 12:09:31,613 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev9
2025-10-31 12:09:31,613 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2025-10-31-12-09-31-mpf-TopGunPinball.log', 'pause': False, 'production': False, 'text_ui': False, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'full_logfile_path': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-10-31-12-09-31-mpf-TopGunPinball.log'}
2025-10-31 12:09:31,613 : INFO : Machine : MPF path: C:\Users\White\mpf\Lib\site-packages\mpf
2025-10-31 12:09:31,613 : INFO : Machine : Machine path: C:\Users\White\Documents\Top_Gun_Pinball
2025-10-31 12:09:31,613 : INFO : Machine : Platform: win32
2025-10-31 12:09:31,613 : INFO : Machine : Python executable location: C:\Users\White\mpf\Scripts\python.exe
2025-10-31 12:09:31,613 : INFO : Machine : Python version: 3.12.7 (64-bit)
2025-10-31 12:09:31,617 : INFO : Machine : Initializing MPF...
2025-10-31 12:09:31,757 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,797 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2025-10-31 12:09:31,797 : INFO : EventManager : Event: ======'machine_var_player1_score'====== Args={'value': 2229000, 'prev_value': None, 'change': True}
2025-10-31 12:09:31,797 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,797 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev9, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,797 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.7', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,952 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-11-10.0.26100-SP0', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,952 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,952 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '11', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,952 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.26100', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,952 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,953 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\White\\Documents\\Top_Gun_Pinball\\logs\\2025-10-31-12-09-31-mpf-TopGunPinball.log', 'prev_value': None, 'change': True}
2025-10-31 12:09:31,953 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2025-10-31 12:09:32,002 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-0', <Platform.Virtual>) <switch.s_trough_1>
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-1', <Platform.Virtual>) <switch.s_trough_2>
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-2', <Platform.Virtual>) <switch.s_trough_3>
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-3', <Platform.Virtual>) <switch.s_trough_4>
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-4', <Platform.Virtual>) <switch.s_trough_5>
2025-10-31 12:09:32,002 : INFO : SwitchController : Found active switch: ('playfield_board_lower-3208-5', <Platform.Virtual>) <switch.s_trough_6>
2025-10-31 12:09:32,039 : ERROR : ConfigValidator : Your config contains a value for the setting "sound_player:p", but this is not a valid setting name.
2025-10-31 12:09:32,039 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...tting name.')>)
handle: <Handle EventManager._queue_task_done(<Task finishe...tting name.')>)>
Traceback (most recent call last):
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\sound_player.py", line 24, in _validate_config_item
    device_settings = self.machine.config_validator.validate_config(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 177, in _validate_config
    self.check_for_invalid_sections(this_spec, source, validation_failure_info)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 345, in check_for_invalid_sections
    raise ConfigFileError('Your config contains a value for the '
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mig_first_letter_collected:sounds with settings {'mig_letter_one_sound': 'play'} >> Your config contains a value for the setting "sound_player:p", but this is not a valid setting name. Error Code: CFE-ConfigValidator-2 (https://missionpinball.org/logs)
2025-10-31 12:09:32,144 : INFO : Machine : Starting the main run loop with active modes: []
2025-10-31 12:09:32,144 : WARNING : Machine : Attract mode is not active, game will not be playable. Please check your attract mode configuration.
2025-10-31 12:09:32,145 : INFO : Machine : Shutting down...
2025-10-31 12:09:32,145 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2025-10-31 12:09:33,160 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\machine.py", line 808, in _run_loop
    raise self._exception['exception']
  File "C:\Users\White\AppData\Local\Programs\Python\Python312\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 795, in _queue_task_done
    future.result()
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\events.py", line 708, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\mode_controller.py", line 97, in initialize_modes
    item.method(config=mode.config[item.config_section],
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_player.py", line 150, in process_mode_config
    config = self.validate_config(config)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_player.py", line 113, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 56, in validate_config_entry
    raise e
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\device_config_player.py", line 52, in validate_config_entry
    self._validate_config_item(device, device_settings))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\config_players\sound_player.py", line 24, in _validate_config_item
    device_settings = self.machine.config_validator.validate_config(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 164, in validate_config
    return self._validate_config(config_spec, source, base_spec=base_spec, add_missing_keys=add_missing_keys,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 177, in _validate_config
    self.check_for_invalid_sections(this_spec, source, validation_failure_info)
  File "C:\Users\White\mpf\Lib\site-packages\mpf\core\config_validator.py", line 345, in check_for_invalid_sections
    raise ConfigFileError('Your config contains a value for the '
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Failed to load config player mig_first_letter_collected:sounds with settings {'mig_letter_one_sound': 'play'} >> Your config contains a value for the setting "sound_player:p", but this is not a valid setting name. Error Code: CFE-ConfigValidator-2 (https://missionpinball.org/logs)
2025-10-31 12:09:33,182 : INFO : root : MPF run loop ended.
